### DISCUSSION TITLE
Suggestion on UI/UX Enhancements

### DISCUSSION THREAD

**Discussion by ShreyashiSoni** [Sunbird-Saral/Community#69] [Ideas]
### Hi all, as per our discussion, updating the UI/UX modifications for the SARAL App (Assessment Use Case Only) to ease out end user (Teacher) experience.

**1. Login Screen**
Pain Point: 
- As a user after logging in, 'Get started' button is not instinctively the next step, which creates a sense of confusion.

Recommendations:
- After login 'Get Started' screen should be replaced with 'Modules' screen, where user can select the 'Use Case' he/she is using the App for.
- After selecting 'Module' user will be directed to 'Fill Exam Details' screen, which will help them instinctively get started with the use case (As per current use case only Assessments)
- Configurable Logo, Configurable Color should be enabled.
- On the Login screen, 'SARAL' App name should be added.

Please refer to attached slide no. 1 & 2
![Slide1](https://user-images.githubusercontent.com/129832399/233544991-2952f740-726f-493f-a673-34d3697544a1.JPG)
![Slide2](https://user-images.githubusercontent.com/129832399/233545002-f3455703-94fd-48cc-abb3-a32295d50299.JPG)

**2. Fill Exam Details Screen**
Pain Points: 
- User was not able to select all subjects, for which the assessment was happening (Limit to only one subject selection)
- Vocabulary Hygiene

Recommendations:
- Write 'School Name' Instead of 'school'
- Remove 'School ID'
- While selecting class nomenclature should be only numeric for example 1,2,3 etc. 
- Enable subjects a combination of subjects based on class for which the assessment is happening
- Write 'Paper Set' instead of 'Set'

Please refer to attached slide no. 3
![Slide3](https://user-images.githubusercontent.com/129832399/233954824-df04644d-8dd5-4247-85fe-54d224f2948c.JPG)

**3. Mark Attendance Screen**
Pain Points:
- As a user, 'mark as absent' or 'mark as present' is not very clearly portraying that all the students are 'Present' by default.
- As a user, scrolling through the long list of students is time taking and tiring (It would be good, if more number of students can be seen on the mobile screen in a single go)
- As a user, after selecting exam details, I should be able to see that information on top, while marking attendance to find any discrepancy/mistake easily.

Recommendations
- Simplify the list view of the students for marking attendance
- Add 'School Name', 'Class' and 'Subject' to left hand corner of screen
- Add Toggle for marking present absent (By default all students to be present)
- Add screen heading 'Mark Attendance'

Please refer to attached slide no. 4
![Slide4](https://user-images.githubusercontent.com/129832399/233954912-d078c4ce-86be-485f-8578-56f64aeb6c1e.JPG)

**4. Home Screen or Summary Page**
Pain Points:
- As a user it is confusing for me to go to next screen for scanning, after marking attendance (Extra step)
- As a user Having two buttons 'Ongoing scan' and 'Scan' is confusing as to which button to use.
- As a user, it was very difficult to find 'Scan' button when using the app first time.
- As a user it is very difficult to keep track of 'Total scanned sheets' and 'Total saved sheets'
- As a user nomenclature of many functionalities are confusing 'Scan Status in summary table', 'Scan status button', 'Dashboard button', 'Ongoing scan button' etc.

Recommendations:
- Remove in between screen with 'Continue Scan' button
- Remove 'Ongoing Scan' button from home screen
- Make scan icon Big and Visible
- Remove 'Exam Type', 'Exam ID', 'Exam Details' from the table
- Remove 'Dashboard' button from the main screen
- Change 'Scan status' button to 'Review Scans' Button
- Change nomenclature as per the below given prototype wherever applicable

Please refer to attached slide no. 5 & 6
![Slide5](https://user-images.githubusercontent.com/129832399/233545094-a38d4003-4c72-424e-9407-28bd8531a2bf.JPG)
![Slide6](https://user-images.githubusercontent.com/129832399/233955282-af9f1eb3-5082-43bb-a08b-4ddf63ce8442.JPG)

**5. During scan messages**
Pain Point:
- As a user it is difficult to read and follow the pop up messages.

Recommendations:
-Make pop up messages more visible by adding a background behind the text

Please refer to attached slide no. 7
![Slide7](https://user-images.githubusercontent.com/129832399/233545137-857bbfa7-3de5-4f72-9dfd-772e1cdb8082.JPG)

**6. Edit and submit scan screen**
Pain Point:
- As a user, I cannot find any functionality of 'Tag' Column, as what should I do with it or what information it will hold.

Recommendations:
- Remove 'tags' column
- Remove 'S.No.' column
- Replace 'Question 1' nomenclature to '1'

Please refer to attached slide no. 8
![Slide8](https://user-images.githubusercontent.com/129832399/233956388-a24e4abd-2c8f-428d-8930-b497d1e11655.JPG)

**7. Review Scans screen**
Pain Points:
- As a user, scrolling through the long list of students is time taking and tiring (It would be good, if more number of students can be seen on the mobile screen in a single go)

Recommendation:
- Make review option simple with the list view

Please refer to the attached slide no. 9
![Slide9](https://user-images.githubusercontent.com/129832399/233955705-2d1f8ad9-a407-479c-a8ef-49cc7a3424eb.JPG)

**8. Reviewing a particular scan**
Pain Point:
- As a user I have to go back to home screen to re-scan and overwrite a particular OMR sheet, if after reviewing I don't find it satisfactory.

Recommendations: 
- Remove 'tag' column
- Add Re-scan functionality incase of some discrepancy, Add 'Re-Scan' button next to 'Close Button'

Please refer to attached slide no. 10
![Slide10](https://user-images.githubusercontent.com/129832399/233545247-5fad2d5e-50e6-4fed-bc8b-47ec70399ad5.JPG)

**9. User Profile Button**
Pain Point: 
As a user the given options are very confusing, and not clearly understandable what is the functionality.

Recommendations:
- Add 'Dashboard' in the profile list
- Review and remove extra options if any in the dropdown.

Please refer to attached slide no. 11
![Slide11](https://user-images.githubusercontent.com/129832399/233545297-2a576645-54f0-4e2c-bfc4-95160fd67f1b.JPG)

Please review above proposed changes and let me know your thoughts for same.
I am also attaching the link to FIGMA file of the created prototypes for better clarity
FIGMA SARAL: [Link](https://www.figma.com/file/alUFLJbFfvqFm13wKD6Vvt/SARaL-UI%2FUX?node-id=0%3A1&t=tXU4Rff1SrUsY5sg-1)

**Comment by ShreyashiSoni** [comment]
@dileep-gadiraju @surendrasinghs

**Reply by karsood** [reply]
Thank you for this detailed post. Truly appreciate it.
I have a questions on some of the pieces that I am sharing below:

1. Saral is not only for assessments. How will the backend identify what saral app will be used for before logging in? What would be the advantage of having this?
2. Why have Subject 1 and Subject 2? What is the use case?
3. Why should 'support' and 'clear user cache' be removed? Where will it be available instead?

**Reply by harish-challa** [reply]
@ShreyashiSoni  Can you please address the above questions. We need a little more details on why these changes i.e subject1 , subject 2  etc. 

We have same list of questions as tried to create user stories.

**Reply by ShreyashiSoni** [reply]
1. Correct Karan! In case one state is using SARAL for more than one use case, we can  add one more screen with option of choosing the use case or enable use case as per login credentials (Teacher, officer etc.). In such case only 'SARAL' will be shown on the front screen, not the use case.
2. Subject 1 and subject 2 was because there are two subjects which gets assessed in the same question paper. This functionality can also be as 'Select Subject', Add subject + (In case there are more than one subjects)
3. Need to understand the use of 'Support', 'Help', 'Clear user cache', 'Clear all user cache' for the user and then we can take a call. keep these extra options with an 'Arrow' to expand the list.

Thanks!

**Comment by harish-challa** [comment]
@ShreyashiSoni also dynamic tagging is enabled from backend, if we disable it from backend, there is no reason to remove from frontend. We can use that option. please check this link : https://saral.sunbird.org/learn/features/dynamic-tagging

**Comment by ShreyashiSoni** [comment]
Hi! I have updated the discussion thread, based on suggestions and discussions. Thanks!

**Comment by navin9584** [comment]
hi @ShreyashiSoni for point number 3 
![Screenshot_2023-04-17-18-37-04-73_f9c6dc39fbf96a0022368795536358b1](https://user-images.githubusercontent.com/91952702/232723565-5ef69302-4799-4710-9573-1b3794fb8f3e.jpg)

**Reply by ShreyashiSoni** [reply]
Hi, Navin

Thank you for sharing this.

I have a few comments on the layout.

1. Please add Block and district details of school. The nomenclature will look like (School Name: Dummy school 4, block name, district name)
2. Please add parent name with the student name. It will look like (Student ID: First Row, Student Name, S/D of Parent Name)
3. Also please keep the background same overall on the screen (Profile button in the background) not above it, and keep the color light or white.

Sorry I missed point 2 in the created mockup

Thanks!

**Reply by karsood** [reply]
HI Shreyashi,

Will adding 1 and 2 increase API calls made?

**Reply by navin9584** [reply]
hi Shreyashi,
Can you also confirm that we can go with this design of present/absent switch

**Reply by harish-challa** [reply]
@karsood @surendrasinghs Adding parent name field, we need to make corresponding backend change as well. Will this be helpful showing in the app?

**Reply by ShreyashiSoni** [reply]
Hi @navin9584 I am okay with the design of present/absent @karsood @surendrasinghs please share your views.

**Reply by ShreyashiSoni** [reply]
> @karsood @surendrasinghs Adding parent name field, we need to make corresponding backend change as well. Will this be helpful showing in the app?

Hi Harish, to make it easy for the teacher to identify the student in case both of the student name is same. It is essential to add parent name as an additional information for verification. As currently in the case of UP, Student ID is not something from which teacher identify student. Also, the detail was there in the previous version when the App was used on ground.

**Reply by ShreyashiSoni** [reply]
> HI Shreyashi,
> 
> Will adding 1 and 2 increase API calls made?

Hi @harish-challa Can you answer this? Will it increase API Calls as well as the time for fetching this information while using online and offline (Downloading master data) or since all of the details are getting fetched from the same database, the efforts are similar. Thanks!

**Comment by ShreyashiSoni** [comment]
@harish-challa @navin9584 Please make sure you are using Figma link to do enhancements, as you can see in detail what are the changes, font type, size etc. Let me know if you need any help.

**Comment by navin9584** [comment]
hi @ShreyashiSoni  i have some query
1. can you plz mail me all images/icon related ui/ux changes

for point screen 5 : The size of the text above the camera cannot be changed, if increased, the text will go out of the mobile screen.
Can change text color if you want plz suggest

thankyou

**Reply by ShreyashiSoni** [reply]
Hi Navin, sharing. 

For Point 5 Screen: The text size is okay! I will share a sample pop up message with you over email. Thanks!
